test/index.test.js

```diff
@@ -0,0 +1,169 @@
+  const { location, repository, organization } = await github.download("SlimIO.Config", {
+    dest: __dirname,
+    branch: "master"
+  });
+  assert.equal(repository, "Config");
+  assert.equal(organization, "SlimIO");
+  assert.equal(location, join(__dirname, "Config-master.tar.gz"));
+
+  await fs.access(location);
+  await fs.unlink(location);
+});
+
+test("download public repository (at current working dir)", async() => {
+  const { location } = await github.download("NodeSecure.utils");
+  assert.equal(location, join(process.cwd(), "utils-main.tar.gz"));
+
+  await fs.access(location);
+  await fs.unlink(location);
+});
+
+test("download private repository (without extraction)", async() => {
+  const { location } = await github.download("SlimIO.Core", {
+    dest: __dirname,
+    branch: "master"
+  });
+  assert.equal(location, join(__dirname, "Core-master.tar.gz"));
+
+  await fs.access(location);
+  await fs.unlink(location);
+});
+
+describe("download public repository (with extraction)", () => {
+  let location;
+
+  before(async() => {
+    const result = await github.downloadAndExtract("SlimIO.is", {
+      dest: __dirname,
+      branch: "master"
+    });
+    location = result.location;
+  });
+
+  test("should have the correct download location", async() => {
+    assert.equal(location, join(__dirname, "is-master"));
+  });
+
+  test("should be a directory", async() => {
+    const st = await fs.stat(location);
+    assert.ok(st.isDirectory());
+  });
+
+  test("should not have the tar.gz file", async() => {
+    const filePath = join(__dirname, "is-master.tar.gz");
+    assert.rejects(
+      async() => await fs.access(filePath),
+      {
+        name: "Error",
+        code: "ENOENT",
+        message: /no such file or directory/
+      }
+    );
+  });
+});
+
+test("download public repository (with extraction and removeArchive disabled)", async() => {
+  const { location } = await github.downloadAndExtract("SlimIO.Safe-emitter", {
+    dest: __dirname,
+    branch: "master",
+    removeArchive: false
+  });
+
+  await fs.access(join(__dirname, "Safe-emitter-master.tar.gz"));
+  assert.equal(location, join(__dirname, "Safe-emitter-master"));
+  assert.equal((await fs.stat(location)).isDirectory(), true);
+});
+
+test("teardown", async() => {
+  await new Promise((resolve) => setTimeout(resolve, 50));
+
+  const dirents = await fs.readdir(__dirname, { withFileTypes: true });
+  for (const dirent of dirents) {
+    if (dirent.name === "index.test.js") {
+      continue;
+    }
+
+    const fullPath = join(__dirname, dirent.name);
+    if (dirent.isDirectory()) {
+      await fs.rm(fullPath, { recursive: true, force: true });
+    }
+    else {
+      await fs.unlink(fullPath);
+    }
+  }
+});
+
+test("get contributors last activites for NodeSecure/scanner", async() => {
+  const contributors = await github.getContributorsLastActivities("NodeSecure", "scanner");
```

Comment:
Seem like there is an issue here (contributors is not a Map so the CI fail with .has)

Reply:
The `getContributorsLastActivities` method returns `null` because of GitHub's ratelimit. So I created a `personal access token` to get around this problem. However, I still get the same error message. On the other hand, when I retrieve the same URL from Insomnia, everything works correctly. I'm continuing my research.

Reply:
Duly noted, I'm looking into it